p4-16/psa/PSA.mdk

```diff
@@ -2576,3 +2576,127 @@ Certainly if there are long durations of time when egress decides to
 drop all packets to an output port, that port will go idle.  The
 scheduling algorithm implementations are all built with a finite
 maximum packet scheduling rate.
+
+
+# Appendix: Multi-pipeline PSA devices
+
+The highest packet rate network devices today are ASICs running at a
+clock rate on the order of 1 to 2 GHz.  This discussion will assume 1
+GHz for the sake of a concrete numerical example, but everything
+discussed here scales linearly with the clock rate.
+
+It is common to design a portion of a network ASIC such that it can
+start processing a new packet once every clock cycle, and finish a
+packet every clock cycle.  The latency might be hundreds of clock
+cycles from starting a packet until it is complete.  P4 tables in such
+ASICs are typically implemented using logic such as TCAMs and SRAMs.
+TCAM designs can do 1 search per clock cycle.  The lowest area and
+power SRAMs can do 1 read or 1 write per clock cycle.
+
+While there are "multi ported" SRAM designs that can be read and/or
+written multiple times per clock cycle, these have a noticeable
+increase in area and power over the "singled ported" designs that are
```

Comment:
Think there's a typo. Should be "single ported".

Reply:
Fixed.